### ISSUE TITLE
Loss modules

### ISSUE THREAD

**Issue by carrascomj** [carrascomj/shapels#29]
Docs: https://docs.pytorch.org/docs/stable/nn.html#loss-functions

Losses subclass the [_Loss](https://github.com/pytorch/pytorch/blob/7716da9fb23f27a65b41f9f016a2afadf281c18f/torch/nn/modules/loss.py#L38). Implementation requires saving to the `TorchNNModule` variant if the reduction is "none". Inference is a Noop if "none", `" "` otherwise, should check if they have the same size (torch will emit a warning if they're not but are broadcastable, or throw an error).

**Comment by carrascomj** [comment]
Inference for losses and their functional counterparts is implemented but is still missing the same-shape check for the two arguments, which should be verified case by case; ~not super sure if~ that's not the case for `CrossEntropyLoss`, for example.
